src/wx/widgets/wx/sdljoy.h

```diff
@@ -20,25 +20,27 @@
 #include "../common/contains.h"
 
 struct wxSDLJoyDev {
-    private:
-        union {
-            SDL_GameController* dev_gc = nullptr;
-            SDL_Joystick*       dev_js;
-        };
-    public:
-        operator SDL_GameController*&();
-        SDL_GameController*& operator=(SDL_GameController* ptr);
+private:
+    union {
+        SDL_GameController* dev_gc = nullptr;
+        SDL_Joystick*       dev_js;
+    };
+public:
+    operator SDL_GameController*&();
+    SDL_GameController*& operator=(SDL_GameController* ptr);
 
-        operator SDL_Joystick*&();
-        SDL_Joystick*& operator=(SDL_Joystick* ptr);
+    operator SDL_Joystick*&();
+    SDL_Joystick*& operator=(SDL_Joystick* ptr);
 
-        operator bool();
+    operator bool();
 
-        std::nullptr_t& operator=(std::nullptr_t&& null_ptr);
+    std::nullptr_t& operator=(std::nullptr_t&& null_ptr);
 };
 
 struct wxSDLJoyState {
     wxSDLJoyDev dev;
+    uint8_t index = 0;
```

Comment:
You should note that the index of the joystick will not remain the same through its lifespan.  For example, if there are two controllers open, and they are assigned index 0 & 1 respectively, and you unplug the controller occupying index 0, the controller that previously sat at index 1 will now sit at index 0 (it shifts down to occupy the lowest index).

Because there is no way for us to track this, we really can't use it in any productive way.

Reply:
The joystick index should be discarded after the connect event and never touched again.  Only instance_id and player_index should be used thereafter.  

From SDL:
```
"The index passed as an argument refers to the N'th game controller on the system. This index is not 
the value which will identify this controller in future controller events. The joystick's instance id 
(SDL_JoystickID) will be used there instead."
```

I think that **player_index** functions the way that you are expecting index to.

From the SDL developer:
```
"Yes, using player_index for binding is the way to go. When connecting a controller it will be assigned 
the first free player index.

Note that if you disconnect and reconnect only 1 controller it will get same player_index. But if you 
disconnect 2 controllers and reconnect them in reverse order they may swap their player_index."
```

Reply:
This is fine, if joystick 0 is configured by the player, it's their problem which joystick is first in their system. The code will work regardless.

Reply:
What do you think will happen in the case of multiple controllers:

Joystick 0, iid 0 and Joystick 1, iid 1 are connected, your map looks like this

joystate[iid 0].index = 0
joystate[iid 1].index = 1

Now, the first player disconnects the controller.

Now, SDL considers iid 1 as index 0.  However, your map considers joystate[iid1].index = 1.  You only update the controller that was disconnected.

Reply:
Also a valid point, I will fix this.

Reply:
OK, I will test some more after your changes.

Reply:
Fixed.